server/src/main/java/org/elasticsearch/cluster/routing/RoutingTable.java

```diff
@@ -499,9 +500,9 @@ public Builder addAsNew(IndexMetaData indexMetaData) {
         }
 
         public Builder addAsRecovery(IndexMetaData indexMetaData) {
-            if (indexMetaData.getState() == IndexMetaData.State.OPEN) {
+            if (indexMetaData.getState() == IndexMetaData.State.OPEN || MetaDataIndexStateService.isIndexMetaDataClosed(indexMetaData)) {
```

Comment:
This is where the distinction is done between a closed index and a closed index on a 8.0 version (with the new index.closed setting)